osnoise-client

```diff
@@ -103,9 +122,39 @@ echo -e "${CMD_OUTPUT}"
 WORKLOAD_CPUS=$(echo -e "${CMD_OUTPUT}" | grep "final cpus:" | awk '{ print $3 }')
 echo "WORKLOAD_CPUS: ${WORKLOAD_CPUS}"
 
-# Run on the workload cpus (exclude hk cpus by default)
-if [ -z ${cpus} ]; then
-    cpus=${WORKLOAD_CPUS}
+# Get the list of house-keeping and update workload if overlaps
+if [ -n $house_keeping ]; then
+    cpu_str=""
+    for cpu in $(echo $house_keeping | sed -e "s/,/ /g"); do
+        cpu_str+=" --cpu $cpu"
+    done
+    # get a comma separated cpu list from house-keeping e.g. 1-3=1,2,3
+    cmd="${TOOLBOX_HOME}/bin/get-cpus-ordered.py --smt ${smt} ${cpu_str}"
+    echo "about to run: ${cmd}"
+    CMD_OUTPUT=$(${cmd})
+    echo -e "${CMD_OUTPUT}"
+    house_keeping=$(echo -e "${CMD_OUTPUT}" | grep "final cpus:" | awk '{ print $3 }')
+
+    # Eliminate overlaps (dup in both workload and house-keeping lists)
+    # remove house-keeping ones from the workload cpus (if any)
+    IFS=',' read -r -a to_remove <<< "$house_keeping"
+    # Loop over each element of to_remove and remove from cpu
+    for cpu in "${to_remove[@]}"; do
+        # Remove the specific number from WORKLOAD_CPUS
+        WORKLOAD_CPUS=$(echo "$WORKLOAD_CPUS" | sed "s/\b$cpu\b//g")
+    done
```

Comment:
Hmm, I'm inclined to say we should error out here if the user does something dumb like include the same CPU in both lists rather than fixing it up for them...otherwise they will never learn.

Reply:
agreed